internal/api/middleware/timer.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-kit/kit/metrics"
+)
+
+func getIdentifier(ctx context.Context) string {
```

Comment:
This is now a nicely encapsulated bit of logic that should be tested. It might be difficult to create a `chi.Context` test value though, I don't see a way for us to put route patterns into it. 

So, I would see if you can find a clean way to create a test context, but if not, then we probably change this func to `getIdentifier(string) string` and have the `Timer` middleware call `RoutePattern()` instead.

Reply:
The timer middleware now calls `RoutePattern` so `getIdentifier` has the suggested signature.